src/libxtp/gaussian_quadrature.cc

```diff
@@ -0,0 +1,187 @@
+              .sum();
+      result += _quadadaptedweights(j) * value;
+    }
+
+  } else if (_opt.quadrature_scheme == "modified_legendre") {
+    // The modified legendre method is suitable for integration limits a = 0 b =
+    // +infty. Here we have value1 and value2 because we split the original
+    // integral from -infty to +infty in two parts. Original legendre quadrature
+    // is meant for integratal with integration limits of -1 and 1. To overcome
+    // this we use the transformation x' = 0.5 ^ (1+x/1-x)
+    for (Index j = 0; j < _opt.order; ++j) {
+      double exponent = (1.0 + _quadpoints(j)) / (1.0 - _quadpoints(j));
+      std::complex<double> newpoint(0.0, std::pow(0.5, exponent));
+      double den =
+          (1.0 - _quadadaptedweights(j)) * (1.0 - _quadadaptedweights(j));
+      double newweight = (2.0 * _quadadaptedweights(j) * 0.5) / den;
+      Eigen::VectorXcd denominator = (DeltaE + newpoint).cwiseInverse() +
+                                     (DeltaE - newpoint).cwiseInverse();
+      std::complex<double> value =
+          ((Imx * (_dielinv_matrices_r[j].conjugate()))
+               .cwiseProduct(denominator.asDiagonal() * Imx))
+              .sum();
+      result += newweight * value;
+    }
+
+  } else if (_opt.quadrature_scheme == "hermite") {
+    // The hermite quadrature method is suitable for integration limits a =
+    // -infty b = +infty. Here we don't do any modification to the original
+    // hermite quadrature method. Points and weights are not transformed
+    for (Index j = 0; j < _opt.order; ++j) {
+      std::complex<double> newpoint(0.0, _quadpoints(j));
+      Eigen::VectorXcd denominator = (DeltaE + newpoint).cwiseInverse();
+      std::complex<double> value =
+          ((Imx * (_dielinv_matrices_r[j]))
+               .cwiseProduct(denominator.asDiagonal() * Imx))
+              .sum();
+      result += _quadadaptedweights(j) * value;
+    }
+
+  } else if (_opt.quadrature_scheme == "legendre") {
+    // This particular legendre quadrature is suitable for integration limits a
+    // = -infty b = +infty. Original Legendre quadrature is meant for
+    // integration limits -1 and +1. The change of variables is x' = tan (pi/2 *
+    // x). When x=-1 we have x'=-infty. When x=1 we have x'=+infty
+    double halfpi = 0.5 * tools::conv::Pi;
+    for (Index j = 0; j < _opt.order; ++j) {
+      std::complex<double> newpoint(0.0, std::tan(halfpi * _quadpoints(j)));
+      double den =
+          std::cos(halfpi * _quadpoints(j)) * std::cos(halfpi * _quadpoints(j));
+      double num = halfpi / den;
+      Eigen::VectorXcd denominator = (DeltaE + newpoint).cwiseInverse();
+      std::complex<double> value =
+          ((Imx * (_dielinv_matrices_r[j]))
+               .cwiseProduct(denominator.asDiagonal() * Imx))
+              .sum();
+      result += num * _quadadaptedweights(j) * value;
+    }
+  }
+  return 0.5 * result.real() / (tools::conv::Pi);
```

Comment:
Why do you take real in the end, you can use `double` all the way? What is the point behind that?